sources/web/datalab/polymer/components/datalab-terminal/datalab-terminal.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/**
+ * Type declaration for xterm.js
+ */
+declare class Terminal {
+  constructor(parameters?: object)
+  open(element: HTMLElement, focus: boolean): void;
+  write(data: string): void;
+  clear(): void;
+  reset(): void;
+  resize(cols: number, rows: number): void;
+  on(event: string, handler: Function): void;
+}
+
+/**
+ * Terminal element for Datalab.
+ * This element uses xterm.js for the front end terminal element, and opens a websocket
+ * connection to a Jupyter terminal on the backend. Although Jupyter supports multiple
```

Comment:
Well the user can still open the dev tools and send an XHR request to create a new terminal, so that case *could* theoretically happen. We don't have to prevent it, but we won't support it.